core/domain/draft_upgrade_services.py

```diff
@@ -226,29 +253,36 @@ def _convert_html_in_draft_change_list(
                 # is allowed to have HTML in the solution correct answer
                 # The typecheckings below can be avoided once #9413 is fixed.
                 if new_value['correct_answer']:
-                    if isinstance(new_value['correct_answer'], list):
+                    # Here we use cast because Solution's correct_answer can be
```

Comment:
I don't think this one is OK -- as usual, you need to cast to an explicit subtype. I also don't really see the condition that forces this to be List[List[str]] -- the only "if" here is whether "new_value['correct_answer']" exists which is not sufficient to determine that.

Remember, to cast, you can only rely on past code (and if conditions), not what happens in the future.

Reply:
We cannot change the type at each isinstane because we are changing the value within the list. But yeah I  have changed the type at last if clause which makes more sense than the previous one PTAL!

Also please specify if you have any other idea!